spec/features/request/admin_crud_internal.steps.rb

```diff
@@ -0,0 +1,119 @@
+require 'spec_helper'
+require 'pry'
+require 'net/http'
+require 'uri'
+
+ICON_IMAGE_NAME = "icon-image.png"
+MAIN_CATEGORY_FORM_MAPPING = {
+  "name" => "Hauptkategorie",
+  "categories.0.name" => "Subkategorien",
+  "categories.0.cost_center" => "Kostenstelle",
+  "categories.0.general_ledger_account" => "Sachkonto ⅠI (IR)",
+  "categories.0.procurement_account" => "Sachkonto I (ER)"
+}
+
+step "I only see the close button" do
+  within find("form[id='#{@request.id}']") do
+    expect(current_scope.all("button").count).to eq 1
+    current_scope.find("button", text: "Schliessen")
+  end
+end
+
+step "I see the link for the request" do
+  within find("form[id='#{@request.id}']") do
+    current_scope.find("a", text: "Link zum Antrag")
+  end
+end
+
+step "I open the AdminKategorien menu" do
+  find("a", text: "Admin").click
+  find("a", text: "Kategorien").click
+end
+
+def wait_for_link_or_throw(text, timeout = 10)
+  sleep_interval = 0.2
+  start_time = Time.current
+
+  loop do
+    begin
+      if page.has_link?(text, wait: 2)
+        puts "Found the link with text: #{text}"
+        break
+      end
+    rescue Selenium::WebDriver::Error::UnexpectedAlertOpenError
+      puts "Handled an unexpected alert."
+    rescue Selenium::WebDriver::Error::NoSuchAlertError
+      puts "No alert was present."
+    end
+
+    if Time.current - start_time > timeout
+      raise "Timeout: Failed to find the link with text '#{text}' within #{timeout} seconds."
+    end
+
+    puts "Sleep #{sleep_interval} seconds"
+    sleep sleep_interval
+  end
+end
```

Comment:
FYI: 
I had to add this "wait_for_link_or_throw"-logic to avoid using of the previous "sleep 2"-logic with fixed timeout.
The nerve-wracking thing is that locally it works fine, but on cider it fails quite easily.

With newer libraries, it should be possible to get rid of the begin/rescue-block - but for me it's not clear which dependencies have to be updated.